Sources/NIO/Selector.swift

```diff
@@ -375,7 +375,8 @@ final class Selector<R: Registration> {
                 let registration = registrations[Int(ev.data.fd)]!
                 try body(
                     SelectorEvent(
-                        readable: (ev.events & Epoll.EPOLLIN.rawValue) != 0 || (ev.events & Epoll.EPOLLERR.rawValue) != 0 || (ev.events & Epoll.EPOLLRDHUP.rawValue) != 0,
+                        readable: (ev.events & Epoll.EPOLLIN.rawValue) != 0,
+                        // We handle EPOLLERR and EPOLLRDHUP only here to ensure we not read if we not really want.
```

Comment:
What's this change for?

Reply:
basically without this we always called `readable()` when a EPOLLRDHUP happened (like connection reset) which then would cause a `channelRead` if there is something on the socket if if the user did not request it.